internal/testing/test-summary/test-summary.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Summarizes the output of go test.
+// Run like so:
+//    go test  -json ./... | test-summary
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+)
+
+// From runing "go doc test2json".
+type TestEvent struct {
+	Time    time.Time // encodes as an RFC3339-format string
+	Action  string
+	Package string
+	Test    string
+	Elapsed float64 // seconds
+	Output  string
+}
+
+func main() {
+	s, err := run(os.Stdin)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(s)
```

Comment:
Should it return a nonzero value if there were failed tests?

Reply:
Done.